### ISSUE TITLE
New attributes: fill-opacity, stroke-opacity

### ISSUE THREAD

**Issue by tkvogt** [diagrams/diagrams-lib#248]
SVG differentiates between fill-opacity and stroke-opacity. It would be handy to have this for the svg parser in diagrams-input. Of course this could be implemented by overlaying two diagrams: One where fill is mempty and one where the outline linewidth is set to 0. But this depends on solving issue [issue 247](https://github.com/diagrams/diagrams-lib/issues/247).

**Comment by cchalmers** [comment]
The fill and stroke opacities can be different by chaining the opacity of the `AlphaColour Double` in their `Texture`.

**Comment by tkvogt** [comment]
Can you give me an example how to set the fill and stroke opacity separately?

**Comment by cchalmers** [comment]
Hmm, if you want to set the stroke opacity for everything in a diagram after you're already set the fill colour, that would be more difficult. If you know the stroke opacity when you're setting the stroke colour it's just ``lcA (c `withOpacity` o)``.

**Comment by byorgey** [comment]
@tkvogt Is this still an issue, or did you figure out a workaround?

**Comment by tkvogt** [comment]
Am 11.09.2015 um 22:59 schrieb Brent Yorgey:
>
> @tkvogt <https://github.com/tkvogt> Is this still an issue, or did you 
> figure out a workaround?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/diagrams/diagrams-lib/issues/248#issuecomment-139660157>.
>
There is no workaround. These attributes are necessary for a perfect 
loading of SVG images. But this only makes every 10th image look bad. 
Could someone of you implement this? I am not so fluent in diagrams 
internals.
It reminds me that I have to finish diagrams-input, mention it in 
diagrams-manual an upload it to hackage.

**Comment by byorgey** [comment]
I don't think this should be too hard to implement.  The hardest part will just be changing backends to support it.  But that can be done incrementally.  I will see if I can get around to this sometime soon.
